File: 1342_number_of_steps_to_reduce_a_number_to_zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        steps = 0
        while num > 0:
            if num % 2 == 0:
                steps += 1
                num /= 2
            else:
                steps += 1
                num -= 1
                
        return steps

File: src/js/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  let step = 0;

  while (num > 0) {
    if (num % 2 === 0) {
      num = num / 2;
      step++;
    } else {
      num--;
      step++;
    }
  }

  return step;
};
